acai-ios/ViewControllers/LoginViewController.swift

```diff
@@ -122,71 +108,191 @@ class LoginViewController: UIViewController {
+            let cell = tableView.dequeueReusableCell(withIdentifier: FileConstants.inputCellReuseIdentifier, for: indexPath) as! InputTableViewCell
+            let inputItem = inputItems[indexPath.row]
+            cell.changeInputTextFieldDelegate = self
+            cell.invalidEntryLabel.isHidden = true
+            if let type = inputItem.type {
+                switch type {
+                case .email:
+                    cell.textField.text = email
+                    if attemptedNetworking {
+                        cell.invalidEntryLabel.isHidden = email.isValidEmail()
+                    }
+                case .firstName:
+                    cell.textField.text = firstName
+                    if attemptedNetworking {
+                        cell.invalidEntryLabel.isHidden = firstName.isValidName()
+                    }
+                case .lastName:
+                    cell.textField.text = lastName
+                    if attemptedNetworking {
+                        cell.invalidEntryLabel.isHidden = lastName.isValidName()
+                    }
+                case .password:
+                    cell.textField.text = password
+                    if attemptedNetworking {
+                        cell.invalidEntryLabel.isHidden = password.isValidPassword()
+                    }
+                case .phoneNumber:
+                    cell.textField.text = phoneNumber
+                    if attemptedNetworking {
+                        cell.invalidEntryLabel.isHidden = phoneNumber.isValidPhoneNumber()
+                    }
+                }
             }
+            cell.configure(for: inputItem)
+            return cell
         }
     }
 
-    @objc func signUp() {
-        if signUpButton.type == .switchDataEntry {
-            loginButton.toggleColor()
-            signUpButton.toggleColor()
-            switchDataEntry(to: .signUp)
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        if indexPath.row == inputItems.count {
+            return FileConstants.buttonCellHeight
         } else {
-            guard let email = emailInputView.textField.text, let name = nameInputView.textField.text, let password = passwordInputView.textField.text else { return }
-            emailInputView.invalidEntryLabel.isHidden = email.isValidEmail()
-            nameInputView.invalidEntryLabel.isHidden = name.isValidName()
-            passwordInputView.invalidEntryLabel.isHidden = password.isValidPassword()
-            if email.isValidEmail() && name.isValidName() && password.isValidPassword() {
-                createUser(name: name, email: email, password: password)
+            return FileConstants.inputViewHeight
+        }
+    }
+
+    func setupKeyboard() {
+        NotificationCenter.default.addObserver(self, selector: #selector(showKeyboard), name: UIResponder.keyboardWillShowNotification, object: nil)
```

Comment:
Can't we achieve this effect with UITextFieldDelegate? NotificationCenter seems overkill for this

Reply:
Also I think if we make this a UITableViewController this keyboard logic will be done for us